### ISSUE TITLE
Changing stack template name causes slowness problems

### ISSUE THREAD

**Issue by usirin** [koding/koding#10554]
Main cause of this is, once we change the title, sidebar tries to render itself all over again.

- We listen to `keyup` event of title input in stack editor.
- We update the title of stack template once a `keyup` event happens, without changing it in backend
- We mark the stack template by setting `isDirty` to `true` to denote that it's not yet updated in backend.
- This updates the title of stack template in relevant `FluxStore`
- Since there is no optimization in `Sidebar` component, it tries to update the whole sidebar.
- Consecutive updates causes ui to not-render efficiently

**tl;dr**: we basically try to update whole `Sidebar` on **every** `keyup` event of `StackEditor`'s name input.

**Comment by usirin** [comment]
@diclee latest sidebar/compute storage changes should fix this issue, we can reopen this if we see this again.
